dali/operators/generic/slice/slice.cc

```diff
@@ -33,6 +33,7 @@ must provide as many dimensions as are specified with the ``axis_names`` or ``ax
 By default, the :meth:`nvidia.dali.ops.Slice` operator uses normalized coordinates and ``WH``
 order for the slice arguments.)code")
     .NumInput(3)
+    .InputDevice(1, 3, InputDevice::CPU)
```

Comment:
I got confused at first with this [1, 3) syntax. Maybe it's just me but `.InputDevice(1, 2, InputDevice::CPU)` would look a bit better here.?

Or even better:
```
.InputDevice(InputDevice::CPU, 1)  // Just the input 2
.InputDevice(InputDevice::CPU, 1, 9)  // From 1 to 9, including 1 and 9
```
In that case I'd go and say explicitly:
```
.InputDevice(InputDevice::MatchBackend, 0)
.InputDevice(InputDevice::CPU, 1)
.InputDevice(InputDevice::CPU, 2)
```
which serves as documentation when you read it

Reply:
The second approach has it's appeal, but InputDox and InputLayout - both take the index as the first element.
I extended it to take a range. And whether it's start, count or begin, end is down to reading the argument names - the "one past" semantics for end index/iterator is rather a common pattern in C++.

Reply:
and there's an overload with 1 argument, so
```.InputDevice(1, InputDevice::CPU)```
is possible